en/upgrade/centreon-ha/upgrade-from-20.04.md

```diff
@@ -0,0 +1,390 @@
+---
+id: upgrade-centreon-ha-from-20-04
+title: Upgrade Centreon HA from Centreon 20.04
+---
+
+This chapter describes how to upgrade your Centreon HA platform from version 20.10
+to version 21.04.
+
+## Prerequisites
+
+### Suspend cluster resources management
+
+In order to avoid a failover of the cluster during the update, it is necessary to unmanage all Centreon resources, as well as MariaDB.
+
+```bash
+pcs resource unmanage centreon
+pcs resource unmanage ms_mysql
+```
+
+### Perform a backup
+
+Be sure that you have fully backed up your environment for the following
+servers:
+
+- Central server
+- Database server
+
+## Upgrade process
+
+### Update the Centreon repository
+
+Run the following commands:
+
+```shell
+yum install -y http://yum.centreon.com/standard/21.04/el7/stable/noarch/RPMS/centreon-release-21.04-4.el7.centos.noarch.rpm
+```
+
+> **WARNING:** to avoid broken dependencies, please refer to the documentation of the additional modules to update the Centreon Business Repositories.
+
+### Upgrade the Centreon solution
+
+Clean yum cache:
+
+```shell
+yum clean all --enablerepo=*
+```
+
+Then upgrade all the components with the following command:
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--HA 2 Nodes-->
+```shell
+yum update centreon\*
+yum install centreon-ha-web centreon-ha-common
+yum autoremove centreon-ha
+```
+<!--HA 4 Nodes-->
+```shell
+# On the Central Servers
+yum update centreon\*
+yum install centreon-ha-web centreon-ha-common
+yum autoremove centreon-ha
+# On the Database Servers
+yum update centreon\*
+yum install centreon-ha-common
+yum autoremove centreon-ha
+```
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+The PHP timezone should be set. Run the command on both Central Server nodes:
+
+```shell
+echo "date.timezone = Europe/Paris" >> /etc/opt/rh/rh-php73/php.d/50-centreon.ini
+```
+
+> **WARNING** the following commands must be executed on only one node of the cluster.
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--HA 2 Nodes-->
+```bash
+pcs resource delete php7
+pcs resource create "php7" \
+    systemd:rh-php73-php-fpm \
+    meta target-role="started" \
+    op start interval="0s" timeout="30s" \
+    stop interval="0s" timeout="30s" \
+    monitor interval="5s" timeout="30s" \
+    clone
+```
+<!--HA 4 Nodes-->
+```bash
+pcs resource delete php7
+pcs resource create "php7" \
+    systemd:rh-php73-php-fpm \
+    meta target-role="started" \
+    op start interval="0s" timeout="30s" \
+    stop interval="0s" timeout="30s" \
+    monitor interval="5s" timeout="30s" \
+    clone
+pcs constraint location php7-clone avoids @DATABASE_MASTER_NAME@=INFINITY @DATABASE_SLAVE_NAME@=INFINITY
```

Comment:
We should avoid using master/slave terms. See https://www.wired.com/story/tech-confronts-use-labels-master-slave/ 

If we do not do it right now, we need to track this for future releases.